src/Ops.Controllers/Areas/Admin/GroupsController.cs

```diff
@@ -0,0 +1,206 @@
+            var paginateModel = new PaginateModel
+            {
+                ItemCount = groupList.Count,
+                CurrentPage = page,
+                ItemsPerPage = filter.Take.Value
+            };
+
+            if (paginateModel.PastMaxPage)
+            {
+                return RedirectToRoute(new
+                    {
+                        page = paginateModel.LastPage ?? 1
+                    });
+            }
+
+            return View(new GroupViewModel
+            {
+                AllGroups = groupList.Data,
+                PaginateModel = paginateModel
+            });
+        }
+        [HttpGet("{groupStub}")]
+        [RestoreModelState]
+        public async Task<IActionResult> Groups(string groupStub)
+        {
+            try
+            {
+                var group = await _groupService.GetGroupByStubAsync(groupStub);
+                group.IsNewGroup = false;
+                group.IsLocationRegion = !string.IsNullOrEmpty(group.SubscriptionUrl);
+                return View("GroupDetails", new GroupViewModel
+                {
+                    Group = group,
+                    Action = nameof(GroupsController.EditGroup)
+                });
+            }
+            catch (OcudaException ex)
+            {
+                ShowAlertDanger($"Unable to find Group {groupStub}: {ex.Message}");
+                return RedirectToAction(nameof(GroupsController.Index));
+            }
+        }
+        [HttpGet("[action]")]
+        [RestoreModelState]
+        public IActionResult CreateGroup()
+        {
+            var group = new Group {
+                IsNewGroup = true
+            };
+
+            return View("GroupDetails", new GroupViewModel
+            {
+                Group = group,
+                Action = nameof(GroupsController.CreateGroup)
+            });
+        }
+
+        [HttpPost]
+        [Route("[action]")]
+        [SaveModelState]
+        public async Task<IActionResult> CreateGroup(Group group)
+        {
+            if (group.IsLocationRegion && string.IsNullOrEmpty(group.SubscriptionUrl))
+            {
+                ModelState.AddModelError("Group.SubscriptionUrl", "A 'Subscription URL' is required for a location region.");
+                ShowAlertDanger($"Invalid paramaters");
+                group.IsNewGroup = true;
+                return RedirectToAction(nameof(CreateGroup));
+            }
+            if (ModelState.IsValid)
+            {
+                try
+                {
+                    await _groupService.AddGroupAsync(group);
+                    ShowAlertSuccess($"Added Group: {group.GroupType}");
+                    group.IsNewGroup = true;
+                    return RedirectToAction(nameof(GroupsController.Groups), new { groupStub = group.Stub });
+                }
+                catch (OcudaException ex)
+                {
+                    _logger.LogError(ex, $"Problem creating Group {group.GroupType}", ex.Message);
```

Comment:
Use [message templates](https://github.com/serilog/serilog/wiki/Writing-Log-Events) when writing to Serilog.